beanie/odm/operators/find/evaluation.py

```diff
@@ -99,7 +100,7 @@ class RegEx(BaseFindEvaluationOperator):
     def __init__(
         self,
         field,
-        pattern: str,
+        pattern: Union[str, re.Pattern]],
```

Comment:
```suggestion
        pattern: Union[str, re.Pattern[AnyStr]],
```
And import `AnyStr` from the "typing" module.

I suggest to also modify the test "test_regex" in tests/odm/operators/find/test_evaluation.py to showcase this new usage:
```python
    pattern = re.compile("^smth")
    q = RegEx(Sample.integer, pattern)
    assert q == {"integer": {"$regex": pattern}}
```